### PR TITLE
DOC: Add user_guide examples and docstring example for df.plot.box and df.plot.hist

### LINKED ISSUE #42492: DOC: Add updated docstring for examples of `by` argument in `df.plot.hist` and `df.plot.box`
This is a follow-up of #28373

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [x] closes #42492
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/user_guide/visualization.rst [hunk 1]
```diff
@@ -448,6 +474,30 @@ columns:
 
     plt.close("all")
 
+You could also create groupings with :meth:`DataFrame.plot.box`, for instance,
```
Comment (jreback) [inline doc/source/user_guide/visualization.rst:477]: can you add a versionchanged 1.4

File: doc/source/user_guide/visualization.rst [hunk 2]
```diff
@@ -316,6 +316,32 @@ The ``by`` keyword can be specified to plot grouped histograms:
    @savefig grouped_hist.png
    data.hist(by=np.random.randint(0, 4, 1000), figsize=(6, 4));
 
+.. ipython:: python
```
Comment (jreback) [inline doc/source/user_guide/visualization.rst:319]: can you add a versionchanged 1.4

File: pandas/plotting/_core.py [hunk 3]
```diff
@@ -1267,6 +1267,16 @@ def box(self, by=None, **kwargs):
             >>> data = np.random.randn(25, 4)
             >>> df = pd.DataFrame(data, columns=list('ABCD'))
             >>> ax = df.plot.box()
+
```
Comment (jreback) [inline pandas/plotting/_core.py:1270]: can you add a versionchanged

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm minor comments

File: doc/source/user_guide/visualization.rst (hunk 1 above)
Comment (charlesdong1991) [reply doc/source/user_guide/visualization.rst]: yep! add for all 3 places!

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @charlesdong1991

**Pull request merged**